### ISSUE TITLE
Capybara

### ISSUE THREAD

**Issue by pachadotdev** [ropensci/software-review#629]
Submitting Author Name: Mauricio Vargas Sepulveda
Submitting Author Github Handle: @pachadotdev
Repository:  https://github.com/pachadotdev/capybara
Submission type: Pre-submission
Language: en

---

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: capybara
Type: Package
Title: Fast and Memory Efficient Fitting of Linear Models With High-Dimensional
    Fixed Effects
Version: 0.2
Authors@R: c(
    person(
        given = "Mauricio",
        family = "Vargas Sepulveda",
        role = c("aut", "cre"),
        email = "m.sepulveda@mail.utoronto.ca",
        comment = c(ORCID = "0000-0003-1017-7574"))
    )
Imports:
    data.table,
    Formula,
    generics,
    MASS,
    stats
Suggests: 
    knitr,
    rmarkdown,
    testthat (>= 3.0.0),
    tidyr
Depends: R(>= 3.5.0)
Description: Fast and user-friendly estimation of generalized linear models with
    multiple fixed effects and cluster the standard errors. The method to obtain
    the estimated fixed-effects coefficients is based on Stammann (2018) 
    <https://arxiv.org/abs/1707.01815> and Gaure (2013)
    <https://dx.doi.org/10.1016/j.csda.2013.03.024>.
License: Apache License (>= 2)
BugReports: https://github.com/pachadotdev/capybara/issues
URL: https://pacha.dev/capybara/, https://github.com/pachadotdev/capybara
LazyData: true
RoxygenNote: 7.3.1
Encoding: UTF-8
NeedsCompilation: yes
LinkingTo: cpp11, cpp11armadillo
VignetteBuilder: knitr
Config/testthat/edition: 3
Remotes: 
    pachadotdev/cpp11armadillo
```

## Scope 

- Please indicate which category or categories from our [package fit policies](https://ropensci.github.io/dev_guide/policies.html#package-categories) or [statistical package categories](https://stats-devguide.ropensci.org/overview.html#overview-categories) this package falls under. (Please check an appropriate box below):

    **Data Lifecycle Packages**

	- [ ] data retrieval
	- [ ] data extraction
	- [ ] data munging
	- [ ] data deposition
    - [ ] data validation and testing
	- [ ] workflow automation
	- [ ] version control
	- [ ] citation management and bibliometrics
	- [ ] scientific software wrappers
	- [ ] field and lab reproducibility tools
	- [ ] database software bindings
	- [ ] geospatial data
	- [ ] text analysis
    
     **Statistical Packages**

	- [ ] Bayesian and Monte Carlo Routines
	- [ ] Dimensionality Reduction, Clustering, and Unsupervised Learning
	- [ ] Machine Learning
	- [x] Regression and Supervised Learning
	- [ ] Exploratory Data Analysis (EDA) and Summary Statistics
	- [ ] Spatial Analyses
	- [ ] Time Series Analyses

- Explain how and why the package falls under these categories (briefly, 1-2 sentences).  Please note any areas you are unsure of:

This helps to estimate linear models with many fixed effects.

- If submitting a statistical package, have you already [incorporated documentation of standards into your code via the **srr** package](https://stats-devguide.ropensci.org/pkgdev.html#pkgdev-srr)?

-   Who is the target audience and what are scientific applications of this package?  

People (mostly) in the social sciences that need multiple controls in their models. This is especially useful in Economics and International Relations.

-   Are there other R packages that accomplish the same thing? If so, how does yours differ or meet [our criteria for best-in-category](https://ropensci.github.io/dev_guide/policies.html#overlap)?

Fixest, alpaca. This one has different design choices and a reduced number of dependencies.

-   (If applicable) Does your package comply with our [guidance around _Ethics, Data Privacy and Human Subjects Research_](https://devguide.ropensci.org/policies.html#ethics-data-privacy-and-human-subjects-research)?

Does not apply.

-  Any other questions or issues we should be aware of?:

No.

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot check srr

**Comment by ldecicco-USGS** [comment]
Hi @pachadotdev 
I'm new to the editor process for the stats submissions, so sorry for the delay.

The stat submission guide is here:
https://stats-devguide.ropensci.org/pkgdev.html

I think the next step for you is to follow the directions here:
https://stats-devguide.ropensci.org/pkgdev.html#pkgdev-srr
https://github.com/ropensci-review-tools/srr

`Capybara` would need srr results to >50% before we can move the package to a full submission.

**Comment by pachadotdev** [comment]
> Hi @pachadotdev I'm new to the editor process for the stats submissions, so sorry for the delay.
> 
> The stat submission guide is here: https://stats-devguide.ropensci.org/pkgdev.html
> 
> I think the next step for you is to follow the directions here: https://stats-devguide.ropensci.org/pkgdev.html#pkgdev-srr https://github.com/ropensci-review-tools/srr
> 
> `Capybara` would need srr results to >50% before we can move the package to a full submission.

i just pushed the results, so far it has 60/116 checks, but I think some do not apply

**Comment by mpadge** [comment]
@pachadotdev There'll always be checks that do not apply to you package. 60/116 is enough to comply, so you could already call `check srr` yourself to confirm. Before doing that, however, could you please indicate how many more you think you may be able to comply with? We intentionally avoid hard limits other than minimal 50%, but it would be better if you could comply with, say, around 60% or more. Do you think that might be possible?

**Comment by pachadotdev** [comment]
> @pachadotdev There'll always be checks that do not apply to you package. 60/116 is enough to comply, so you could already call `check srr` yourself to confirm. Before doing that, however, could you please indicate how many more you think you may be able to comply with? We intentionally avoid hard limits other than minimal 50%, but it would be better if you could comply with, say, around 60% or more. Do you think that might be possible?

yes, if I move the scripts in dev to formal tests, it would be 70% i would say

**Comment by ldecicco-USGS** [comment]
@ropensci-review-bot check srr

**Comment by ldecicco-USGS** [comment]
@pachadotdev let me know (or run the review-bot command to check srr) when you've moved those scripts in dev to to formal tests.

**Comment by ldecicco-USGS** [comment]
Thank you @pachadotdev for this pre-submission inquiry. This package is within our statistical scope and we'd be happy to review a full submission. You'll need to open a new issue for the full submission. 

(see issue 615 for a recent example of a submission, where the pre-submission is 614)
